loom/graphics/gfxQuadRenderer.cpp

```diff
@@ -99,28 +92,12 @@ void QuadRenderer::submit()
             
             if (!sShaderStateValid)
             {
-                // Select the shader.
-                Graphics::context()->glUseProgram(sProgramPosColorTex);
-
-                Graphics::context()->glEnableVertexAttribArray(sProgram_posAttribLoc);
-                Graphics::context()->glEnableVertexAttribArray(sProgram_posColorLoc);
-                Graphics::context()->glEnableVertexAttribArray(sProgram_posTexCoordLoc);
-            
-                Graphics::context()->glVertexAttribPointer(sProgram_posAttribLoc,
-                                                           3, GL_FLOAT, false,
-                                                           sizeof(VertexPosColorTex), (void*)offsetof(VertexPosColorTex, x));
-
-                Graphics::context()->glVertexAttribPointer(sProgram_posColorLoc,
-                                                           4, GL_UNSIGNED_BYTE, true,
-                                                           sizeof(VertexPosColorTex), (void*)offsetof(VertexPosColorTex, abgr));
-
-                Graphics::context()->glVertexAttribPointer(sProgram_posTexCoordLoc,
-                                                           2, GL_FLOAT, false,
-                                                           sizeof(VertexPosColorTex), (void*)offsetof(VertexPosColorTex, u));
-
-                Graphics::context()->glUniform1i(sProgram_texUniform, 0);
-                Graphics::context()->glUniformMatrix4fv(sProgram_mvp, 1, GL_FALSE, Graphics::getMVP());
-
+                Loom2D::Matrix mvp;
+                mvp.copyFromMatrix4(Graphics::getMVP());
+                sCurrentShader->setMVP(mvp);

```

Comment:
No, this sets the property, the example then reads this property and sets the uniform value.